### ISSUE TITLE
[valid-describe-callback] rule does not allow more than just function literals

### ISSUE THREAD

**Issue by MarcellPerger1** [jest-community/eslint-plugin-jest#1185]
Something like this:
```js
function testAddition() {
  it("Adds 1 + 1", () => {
    expect(1 + 1).toStrictEqual(2);
  })
}

describe("addition", testAddition);
```
fails with:
```
  7:10  error  Second argument must be function  jest/valid-describe-callback
```
But `testAddition` is a function so why does it fail?

**Comment by G-Rath** [comment]
> `testAddition` is a function so why does it fail?

Because we cannot accurately know that it is a function as we're not TypeScript - while in the example you gave it would be trivial for us to walk the AST tree and determine that `testAddition` is in fact a function, that's not commonly how this is used in real world code because there's no real gain vs just inlining the function as a callback.

What could be more useful is to either have a common test function that is defined in a helper file and exported:

```
import { accessibilityTests } from './shared';

describe("accessibility",  accessibilityTests);
```

Or to have a partial function:

```
const testObject = objBuilder => {
  it('must have at least one property', () => {
    expect(Object.keys(objBuilder()).length).toBeGreaterThan(0);
  });

  it('must be stringifable', () => {
    expect(() => JSON.stringify(objBuilder())).not.toThrow();
  });
};

describe('first object', testObject(() => ({ hello: 'world' }));
describe('second object', testObject(() => ({ hello: 'sunshine' }));
```

In both cases we can't accurately confirm that you are passing a function that matches the expected signature of `describe`, and there's also two better ways you can structure these that'll make them work better with `eslint-plugin-jest`: either include the `describe` in your function or call your function within the callback:

```
function testAddition() {
  describe("addition", () => {
    it("Adds 1 + 1", () => {
      expect(1 + 1).toStrictEqual(2);
    })
  });
}

testAddition();
```

or

```
function testAddition() {
  it("Adds 1 + 1", () => {
    expect(1 + 1).toStrictEqual(2);
  })
}

describe("addition", () => {
  testAddition());
});
```
